File: tests/integration/construct_component_unknown/dotnet/Random.cs
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

// Exposes the Random resource from the testprovider.
// Requires running `make test_build` and having the built provider on PATH.

using Pulumi;

public partial class Random : Pulumi.CustomResource
{
	[Output("length")]
	public Output<int> Length { get; private set; } = null!;

	[Output("result")]
	public Output<string> Result { get; private set; } = null!;

	public Random(string name, RandomArgs args, CustomResourceOptions? options = null)
		: base("testprovider:index:Random", name, args ?? new RandomArgs(), options)
	{
	}
}

public sealed class RandomArgs : Pulumi.ResourceArgs
{
	[Input("length", required: true)]
	public Input<int> Length { get; set; } = null!;

	public RandomArgs()
	{
	}
}

File: tests/integration/construct_component_unknown/go/random.go
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

// Exposes the Random resource from the testprovider.
// Requires running `make test_build` and having the built provider on PATH.

package main

import (
	"reflect"

	"github.com/pkg/errors"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

type Random struct {
	pulumi.CustomResourceState

	Length pulumi.IntOutput    `pulumi:"length"`
	Result pulumi.StringOutput `pulumi:"result"`
}

func NewRandom(ctx *pulumi.Context,
	name string, args *RandomArgs, opts ...pulumi.ResourceOption) (*Random, error) {
	if args == nil || args.Length == nil {
		return nil, errors.New("missing required argument 'Length'")
	}
	if args == nil {
		args = &RandomArgs{}
	}
	var resource Random
	err := ctx.RegisterResource("testprovider:index:Random", name, args, &resource, opts...)
	if err != nil {
		return nil, err
	}
	return &resource, nil
}

type randomArgs struct {
	Length int `pulumi:"length"`
}

type RandomArgs struct {
	Length pulumi.IntInput
}

func (RandomArgs) ElementType() reflect.Type {
	return reflect.TypeOf((*randomArgs)(nil)).Elem()
}
